miio/gateway_scripts.py

```diff
@@ -0,0 +1,324 @@
+from json import dumps as dumps_orig
+from random import randint
+
+separators = (",", ":")
+
+
+def dumps(data):
+    return dumps_orig(data, separators=separators)
+
+
+fake_device_id = "120009025"
+fake_device_model = "chuangmi.plug.v3"
+
+
+def sid_to_num(sid):
+    lumi, hex_id = sid.split(".")
+    num_id = int.from_bytes(bytes.fromhex(hex_id), byteorder="big")
+    return str(num_id)[-6:]
+
+
+action_prefix = "x.scene."
+
+# Keeping action script tail decimal int it might be used as index in some db
+action_id = {
+    "move": lambda sid: action_prefix + "1" + sid_to_num(sid),
+    "rotate": lambda sid: action_prefix + "2" + sid_to_num(sid),
+    "singlepress": lambda sid: action_prefix + "3" + sid_to_num(sid),
+    "doublepress": lambda sid: action_prefix + "4" + sid_to_num(sid),
+    "shake": lambda sid: action_prefix + "5" + sid_to_num(sid),
+    "longpress": lambda sid: action_prefix + "6" + sid_to_num(sid),
+    "flip90": lambda sid: action_prefix + "7" + sid_to_num(sid),
+    "flip180": lambda sid: action_prefix + "8" + sid_to_num(sid),
+    "shakeair": lambda sid: action_prefix + "9" + sid_to_num(sid),
+    "taptap": lambda sid: action_prefix + "10" + sid_to_num(sid),
+}
+
+
+def _inflate(
+    action,
+    extra,
+    source_sid,
+    source_model,
+    target_id,
+    target_ip,
+    target_model,
+    target_encoded_token,
+    event=None,
+    command_extra="",
+):
+    if event is None:
+        event = action
+
+    lumi, source_id = source_sid.split(".")
+
+    return [
+        [
+            action_id[action](source_sid),
+            [
+                "1.0",
+                randint(1590161094, 1590162094),
+                [
+                    "0",
+                    {
+                        "did": source_sid,
+                        "extra": extra,
+                        "key": "event." + source_model + "." + event,
+                        "model": source_model,
+                        "src": "device",
+                        "timespan": ["0 0 * * 0,1,2,3,4,5,6", "0 0 * * 0,1,2,3,4,5,6"],
+                        "token": "",
+                    },
+                ],
+                [
+                    {
+                        "command": target_model + "." + action + "_" + source_id,
+                        "did": target_id,
+                        "extra": command_extra,
+                        "id": randint(0, 999),
+                        "ip": target_ip,
+                        "model": target_model,
+                        "token": target_encoded_token,
+                        "value": "",
+                    }
+                ],
+            ],
+        ]
+    ]
+
+
+def build_move(
+    source_sid,
+    target_ip,
+    target_encoded_token,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_cube.v1",
```

Comment:
Considering that these seem to be passed everywher, I think the best approach is to create a new class, `EventHandler` could be a possible name. This would contain all the necessary information for the handler, so just to give you an idea:
```
@attr.s
class EventHandler:
    source_sid = attr.ib()
    target_ip = attr.ib()
    target_token = attr.ib()
    target_model = attr.ib()  # is this necessary?!
    target_did = attr.ib()  # should this be `target_sid`?
    target_model = attr.ib() # is this necessary?
```
Potentially a lot of these are already known by the target/source devices, so potentially passing the source "device" and target "device" should be enough for making the connections.